src/views/credentials/addCredentialModal.tsx

```diff
@@ -53,28 +70,46 @@ const useCredentialForm = (credentialType: string | undefined, credential?: Cred
   };
   const [formData, setFormData] = useState<CredentialFormType>(initialFormState);
 
-  const typeValue = credential?.cred_type || credentialType?.split(' ')?.shift()?.toLowerCase() || '';
+  const typeValue = credential?.cred_type || credentialType?.split(' ')?.shift()?.toLowerCase();
   const [authType, setAuthType] = useState('');
 
+  const deriveAuthType = () => {
+    if (credential) {
+      return helpers.getAuthType(credential);
+    }
+    switch (typeValue) {
+      case 'openshift':
+      case 'rhacs':
+        return 'Token';
+      default:
+        return 'Username and Password';
+    }
+  };
+
   useEffect(() => {
-    // this could also be a helper, for testing
-    const deriveAuthType = () => {
-      switch (typeValue) {
-        case 'openshift':
-        case 'rhacs':
-          return 'Token';
-        default:
-          return 'Username and Password';
-      }
-    };
+    const maskIfTrue = hasValue => (hasValue ? '****' : '');
+
+    if (credential) {
+      setFormData({
+        password: maskIfTrue(credential?.has_password),
```

Comment:
Testing the actual build and it appears that indeed, password fields I did not interact with are overwritten with `****`. I'm afraid this would be a blocker for this PR.

Reply:
I tested 7326ab3dd6fdb0dc63b53ed96e28decfbb54c762 and this is mostly fixed. If I change some other field during editing and click Save (without ever focusing password field), password is not overwritten in database. In fact, I can focus password field and move to another field (i.e. when using Tab to switch between form fields), and it will still not be overwritten. Which is nice, because previous version forced me to re-type my password in such case.

However, there is one consequence of using placeholder that we have not thought through - there's no way to clear **optional** password field. Imagine following scenario:

1. I create new credential of network type. I put SSH key and SSH passphrase.
2. Now I realize that this key does not have a passphrase. So I edit the credential.
3. But SSH passphrase is already "empty". If I type anything and remove it, it will revert to placeholder value which means "just keep using old value". There's no way for me to set field to actual empty string.

This only applies to "SSH passphrase" and "Become Password" fields, as these are only two optional password fields. I don't have good idea how to fix that.

Reply:
@mirekdlugosz good catch...  @nicolearagao i'll take a look at making them optional. with the recent update it's easier to handle now... 

... the solution is to just remove those 2 fields from the list =)

Reply:
After looking at this a little bit more, yes, technically we could allow the "optional" fields to pass through, but shouldn't, apologies on thinking this would be quick =) 

Mireks 3rd point is spot on, a user would expect that backing out the edited field should restore the original value... that being said in the past we've seen solutions that

- involve adding a checkbox, button, or toggle reset for the field that indicates to the user it should be reset

This would involve more effort than expected

Reply:
I created [DISCOVERY-860](https://issues.redhat.com/browse/DISCOVERY-860) to capture this specific problem of clearing optional password fields.